client/my-sites/stats/all-time-highlights-section/index.tsx

```diff
@@ -151,13 +154,25 @@ export default function AllTimeHighlightsSection( {
 		};
 	}, [ isStatsLoading, translate, views, viewsBestDay, viewsBestDayTotal, userLocale ] );
 
+	// TODO: Investigate granularity on Insights page.
+	// Do we want to get granular on this page? It's probably not necessary yet.
+	const shouldLockCards = useShouldGateStats( STAT_TYPE_INIGHTS );
+
 	const highlightCardsMobile = (
 		<div className="highlight-cards-mobile">
 			<h3 className="highlight-cards-heading">{ translate( 'Highlights' ) }</h3>
 			<DotPager>
-				<AllTimeStatsCard infoItems={ infoItems } />
-				<MostPopularDayTimeCard cardInfo={ mostPopularTimeItems } />
-				<MostPopularDayTimeCard cardInfo={ bestViewsEverItems } />
+				<AllTimeStatsCard infoItems={ infoItems } siteId={ siteId } isLocked={ shouldLockCards } />
+				<MostPopularDayTimeCard
+					cardInfo={ mostPopularTimeItems }
+					siteId={ siteId }
+					isLocked={ shouldLockCards }
+				/>
+				<MostPopularDayTimeCard
+					cardInfo={ bestViewsEverItems }
+					siteId={ siteId }
+					isLocked={ shouldLockCards }
+				/>
```

Comment:
This code is the same in both the desktop & mobile versions of the JSX. I tried factoring it out (as a const element) but that breaks the `DotPager` setup and causes the cards to display inline instead of in the slider UI.

ie: The following did not work in my testing:

```
<DotPager>{ innerCardElements }</DotPager>
```

I'd be interested to hear any tips for how we could refactor this repreated code block.

Reply:
Something like this should work:

```JavaScript
const Aa = () => null;

const AaArray = [ <Aa key={ 1 } />, <Aa key={ 2 } />, <Aa key={ 3 } /> ];

function Bb() {
	return <div>{ ...AaArray }</div>;
}

```

Reply:
PS: `<></>` seems to be empty wrapper, but seeing from the`DotPager` side, it would be one single element rather than an array.

Meaning if you pass the following to a wrapper, `children` would be an object:

```JavaScript
<>
<X /> <Y /> <Z />
</>
```

But array in the following case:

```
<X /> <Y /> <Z />
```

https://github.com/Automattic/wp-calypso/blob/8e128699b8eaaddf37ad0ce6daeec4ac6d3bbca6/packages/components/src/swipeable/index.js#L81

PS: my approach mentioned in the previous comment wouldn't work as React doesn't support spreading components unfortunately.

Reply:
`highlightCardsStandard` and `highlightCardsMobile` are not that different it seems. We only need to toggle the className and add dot pager for mobile view. I guess I would go in this direction instead.

Also you could choose to extract `DotPage` with the section, and use a Fragment for the desktop version. Basically extract more rather than only the list of cards I suppose.